app/MMB/src/scenes/tripServices/transportation/TransportationMap.js

```diff
@@ -0,0 +1,350 @@
+    ) {
+      navigation.setParams({ disabled: false });
+    }
+  }
+
+  openLocationPicker = async () => {
+    const { currentLocation } = this.state;
+    let currentAddress;
+
+    if (currentLocation != null) {
+      const { latitude, longitude } = currentLocation;
+      currentAddress = await this.getFormattedAddress({
+        latitude,
+        longitude,
+      });
+    }
+
+    this.props.navigation.navigate('AddressPickerScreen', {
+      currentAddress,
+    });
+  };
+
+  openSettings = () => {
+    Linking.openURL('app-settings:');
+  };
+
+  alertOpenSettings = () => {
+    Alert.translatedAlert(
+      undefined,
+      {
+        id: 'mmb.trip_services.transportation.map.current_position_alert',
+      },
+      [
+        {
+          text: { id: 'mmb.alert.button.ok' },
+          undefined,
+          style: 'default',
+        },
+        {
+          text: { id: 'mmb.alert.button.settings' },
+          onPress: this.openSettings,
+          style: 'default',
+        },
+      ],
+    );
+  };
+
+  onRegionChange = (region: Region) => {
+    this.setState({ region });
+  };
+
+  buildWhitelabelURL = (whitelabelURL: string) => {
+    const { markerA, markerB } = this.state.markers;
+    if (markerA != null && markerB != null) {
+      const URLbase = whitelabelURL.replace(/\?.*/, '');
+      const URLparameters = whitelabelURL.replace(/.*\?/, '');
+
+      const parameters = querystring.parse(URLparameters);
+
+      const pickup = `${markerA.latitude},${markerA.longitude}`;
+      const dropoff = `${markerB.latitude},${markerB.longitude}`;
+      const language = DeviceInfo.getLanguage() || 'en-gb';
+      const currency = 'EUR';
+
+      parameters.pickup = pickup;
+      parameters.dropoff = dropoff;
+      parameters.language = language;
+      parameters.currency = currency;
+
+      return URLbase + '?' + querystring.stringify(parameters);
+    }
+  };
+
+  openLink = () => {
+    const { markerA, markerB } = this.state.markers;
+    if (markerA != null && markerB != null) {
+      const whitelabelURL = this.props.params.whitelabelURL;
+      const url = this.buildWhitelabelURL(whitelabelURL);
+
+      this.props.navigation.navigate('mmb.trip_services.webview', {
+        url,
+      });
+    }
+  };
+
+  renderMarkerA = (e: NativeEvent) => {
+    const { latitude, longitude } = e.nativeEvent.coordinate;
+    this.setState(state => ({
+      markers: {
+        ...state.markers,
+        markerA: {
+          latitude,
+          longitude,
+        },
+      },
+    }));
+  };
+
+  renderMarkerB = (e: NativeEvent) => {
+    const { latitude, longitude } = e.nativeEvent.coordinate;
+    this.setDestination({ latitude, longitude });
+    this.setState(state => ({
+      markers: {
+        ...state.markers,
+        markerB: {
+          latitude,
+          longitude,
+        },
+      },
+    }));
+  };
+
+  getFormattedAddress = async ({ latitude, longitude }: Coordinate) => {
+    const address = await fetch(
```

Comment:
please add this to graphql, all calls should go through there

Reply:
Ok, I will add it there.

Reply:
Decided against it, for now.